Create new group views

Tai-f opened the pull request:
#What
1. generate new-group-view
2. configure routes
3. generate "create" method to save the  values as a new group

#Why
to create new-group-view (one of the main function)

Commit:
create pull request

Commit:
Add group materials

1. add group materials
( group.scss, new.html.haml, edit.html.haml)
2. convert erb. to .haml
 (delete erb.files)

Commit:
Apply group.scss

add @import “group”; to application.scss

Commit:
Add group_id to messages table

1. add group_id to messages table
2. configure the foreign_key constraint in messages table
3.

Commit:
Configure routes of "groups"

Add “create” action of groups

Commit:
Define "create" method

1. define “create” method
2. configure strong parameters

Commit:
Edit new.html.haml

edit new.html.haml
 to link “create”
to save value of “name” in groups.

hand12 commented on app/controllers/groups_controller.rb:
```diff
@@ -1,4 +1,14 @@
+    redirect_to controller: :messages, action: :index
+  end
+
+  protected
```
protectedは、GroupsControllrクラスのインスタンスと、このクラスの中以外からは呼び出されないようにするものですが、似たようなものでprivateというものがあります。
こちらはこのメソッドを定義しているクラスの中でのみ呼び出すことが可能となるもので、インスタンスからは呼び出せません。
今回の場合、インスアンスから呼び出しているわけではないので、privateを使うので大丈夫です！
http://qiita.com/tbpgr/items/6f1c0c7b77218f74c63e
こちらのサイトにも載っていますので一度見てください！

Tai-f replied:
了解です！

hand12 commented on app/controllers/groups_controller.rb:
```diff
@@ -1,4 +1,14 @@
+
+  protected
+    def create_params
+      params.permit( :name)
```
requireも使いましょう！
requireメソッドが何をしているのかと、このstrongparameterはなんのために行っているのかも調べて見てください！

hand12 commented on app/views/devise/unlocks/new.html.haml:
```diff
@@ -0,0 +1,10 @@
+    = f.email_field :email, autofocus: true
+  .actions
+    = f.submit "Resend unlock instructions"
+= render "devise/shared/links"
```
deviseでのuserモデルを作成するのであれば、deviseの導入や、userモデルの導入のブランチでdevioseの返還も行ってください！
ブランチ名と関係ないことは基本的に避けてください。

Tai-f replied:
間違えてcommitしてしまったようです。このファイルは触っていないので、spaceなどで反映されてしまった可能性があります。以後気をつけます。

hand12 commented on app/views/groups/edit.html.haml:
```diff
@@ -0,0 +1,25 @@
+    .chat-group-form__field.clearfix
+      .chat-group-form__field--left
+      .chat-group-form__field--right
+        %input.chat-group-form__action-btn{"data-disable-with" => "Save", :name => "commit", :type => "submit", :value => "Save"}/
```
Groupの編集昨日も入れるのであれば、ブランチ名はGroup機能の追加などにしておきましょう！
また、inputタグはform_forに直してください！

Tai-f replied:
こちらも同様にspaceなどで、反映してしまったようです。commitする際に注意深くindexに送るファイルを確認します。

hand12 commented on app/views/groups/new.html.haml:
```diff
@@ -0,0 +1,29 @@
+    .chat-group-form__field.clearfix
+      .chat-group-form__field--left
+      .chat-group-form__field--right
+        %input.chat-group-form__action-btn{"data-disable-with" => "Save", :name => "commit", :type => "submit", :value => "Save"}/
```
こちらもですね！

hand12 commented on db/migrate/20161118083706_create_messages.rb:
```diff
@@ -3,6 +3,7 @@ def change
     create_table :messages do |t|
       t.text         :body
       t.string       :image
+      t.references   :group, foreign_key: true
```
indexも貼りましょう！

Tai-f replied:
了解です！

hand12 reviewed (commented):
いくつかコメントしたのでご確認ください！

Commit:
Change visibility

change visibility
From “protected” to “private”

Commit:
Change migration files

1. add index to group_id in messages table
2. edit files to avoid errors

Commit:
Remove group_id_id from messages table

Remove group_id_id from messages table because the spell is not
appropriate  and it already exists “group_id” in messages table.

Commit:
Edit groups_controller.rb

1.  define @group in “new”method
2. add require( :group) in create_params method

Commit:
Convert from "form"tag to "form_for"

hc0208 commented on app/controllers/groups_controller.rb:
```diff
@@ -1,4 +1,15 @@
   end
+
+  def create
+    Group.create(create_params)
```
group名が空の時は保存されないようにモデルにバリデーションをかけましょう。
コントローラーにはgroupの作成に成功した時と失敗した時の処理を書くようにしましょう。

Tai-f replied:
validateについては記述しましたが、コントローラーについての記述がわかりません。
ご教授いただければ幸いです。

Commit:
Validates name of group model

it validates name in group model

Commit:
Edit groups_controller.rb

add “if” method

hc0208 commented on app/controllers/groups_controller.rb:
```diff
@@ -1,4 +1,19 @@
   end
+
+  def create
+    group = Group.create(create_params)
```
@group = Group.new(create_params)
としてあげて@group.saveが成功するのかしないのかで条件分岐してあげるのがベストです。

Tai-f replied:
了解しました。

Commit:
Modify groups_controller.rb

add if method by @group.save

hc0208 commented:
LGTM
